kv/replica_slice.go

```diff
@@ -112,6 +114,8 @@ func (rs replicaSlice) SortByCommonAttributePrefix(attrs []string) int {
 				// This packs all matching replicas together.
 				rs.Swap(firstNotOrdered, i)
 				firstNotOrdered++
+			} else {
```

Comment:
is this the right place? I would've expected this before l121 below; that's where you know that everything including and after `firstNotOrdered` isn't going to be touched any more and constitutes a bucket which should be shuffled.